tests/components/picnic/test_sensor.py

```diff
@@ -312,6 +314,25 @@ async def test_sensors_last_order_in_future(self):
             "sensor.picnic_last_order_eta_end", "2021-02-26T20:14:00+00:00"
         )
 
+    async def test_sensors_eta_date_malformed(self):
+        """Test sensor states when last order eta dates are malformed."""
+        # Set-up platform with default mock responses
+        await self._setup_platform(use_default_responses=True)
+
+        # Set non-datetime strings as eta
+        eta_dates: Dict[str, str] = {
+            "start": "wrong-time",
+            "end": "other-malformed-datetime",
+        }
+        delivery_response = copy.deepcopy(DEFAULT_DELIVERY_RESPONSE)
+        delivery_response["eta2"] = eta_dates
+        self.picnic_mock().get_deliveries.return_value = [delivery_response]
+        await self._coordinator.async_refresh()
+
+        # Assert eta times are not available due to malformed date strings
+        self._assert_sensor("sensor.picnic_last_order_eta_start", STATE_UNAVAILABLE)
+        self._assert_sensor("sensor.picnic_last_order_eta_end", STATE_UNAVAILABLE)
```

Comment:
Unavailable? It should be unknown?

Unavailable should be a result of the service being unreachable (not unexpected values).

Reply:
Have now changed it to unknown!